server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -484,7 +484,7 @@ protected Node(
                 for (ExtensionAwarePlugin extAwarePlugin : extensionAwarePlugins) {
                     additionalSettings.addAll(extAwarePlugin.getExtensionSettings());
                 }
-                this.extensionsManager = new ExtensionsManager(additionalSettings);
+                this.extensionsManager = new ExtensionsManager(additionalSettings, identityService);
```

Comment:
I think the constructor is the way to go, but I do see that ExtensionsManager has setters (albeit unused) for other services like this: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/extensions/ExtensionsManager.java#L490-L492

Reply:
FYI, the setters were (are still?) there because of the order in which we initiated, a chicken-and-egg sort of a thing.  I agree we shouldn't use them if we don't need them.

Reply:
This seems out of scope for the current change as it isn't making the problem _worse_